kratos/mpi/includes/mpi_data_communicator.h

```diff
@@ -383,12 +393,25 @@ class KRATOS_API(KRATOS_MPI_CORE) MPIDataCommunicator: public DataCommunicator
     ///@name Member Variables
     ///@{
 
+    /// The MPI communicator
     MPI_Comm mComm;
 
+    /// The list of ranks involved in this DataCommunicator
+    mutable std::vector<int> mRanksList;
+
     ///@}
     ///@name Operations
     ///@{
 
+    /**
+     * @brief Generate a rank list using the current rank and size of the system.
+     * @details This function generates a rank list by first obtaining the current rank and size
+     * of the system, and then using MPI's AllGather function to gather the ranks of all
+     * processes into a vector.
+     * @return A std::vector<int> containing the rank list of all processes.
+     */
+    std::vector<int> GenerateRankList() const;
```

Comment:
```suggestion
    std::vector<int> GenerateRankList();
```

Reply:
i will try, but I think I already tried and I couldn't use it as ususally the data communicators are const

Reply:
but thats used only in the CTor no?

Reply:
True, I have change it